### PR TITLE
Fix Windows build

### PR DESCRIPTION
**Pull request by jorisv**
- [x] clang-cl build
- [x] cl build
- [x] Fix TU
- [x] CI

### CODE REVIEW

**Comment by jorisv** [comment]
I have test on Windows with clang and clang-cl.

There is two tests that are not working right now:
-  curves_tests: Exception
-  test-s3-smooth: precision error and maybe bad stream input

@nim65s Do you think the pip workflow can work on Windows ?

**Comment by nim65s** [comment]
cmeel does not support windows, because I have no idea how this platform is supposed to work :/

**Comment by jorisv** [comment]
Ok, I can add a conda/pixi workflow if you think it's worth it.
There is also the conda-forge CI that can check Windows regression when a new release is done.

**Comment by nim65s** [comment]
It's up to you :)

**Comment by jorisv** [comment]
It's all good.
The recipe build well on Windows with cl and clang-cl.
I have added a Windows CI with conda.

File: .github/workflows/windows-conda.yml [hunk 1]
```diff
@@ -0,0 +1,64 @@
+name: Build ndcurves for Windows via conda
+
+on: [push, pull_request]
```
Comment (nim65s) [inline .github/workflows/windows-conda.yml]:
```suggestion
on:
  push:
    branches:
      - devel
      - master
  pull_request:
    branches:
      - devel
      - master
```

This should avoid the workflow to be triggered twice

File: tests/test-so3-smooth.cpp [hunk 2]
```diff
@@ -73,7 +73,7 @@ BOOST_AUTO_TEST_CASE(from_quat_and_time_constructor) {
 BOOST_AUTO_TEST_CASE(from_quat_and_time_generate) {
   // Create some start variables.
   quaternion_t init_quat = quaternion_t::UnitRandom();
-  matrix3_t init_rot = init_quat.toRotationMatrix();
+  matrix3_t init_rot = init_quat.normalized().toRotationMatrix();
```
Comment (nim65s) [inline tests/test-so3-smooth.cpp:76]: `UnitRandom()` does not provide a unitary quaternion ?

Comment (jorisv) [reply tests/test-so3-smooth.cpp:76]: It should but `SO3Smooth::generate` renormalize the quaternion :

```cpp
  void generate(const quaternion_t& init_rot, const quaternion_t& end_rot,
                const time_t t_min, const time_t t_max) {
    quaternion_t init_rot_normalized = init_rot.normalized();
    quaternion_t end_rot_normalized = end_rot.normalized();
    generate(init_rot_normalized.toRotationMatrix(),
             end_rot_normalized.toRotationMatrix(), t_min, t_max);
  }
```

I think this normalization change a little bit the quaternion value and make this check fail :

```cpp
BOOST_CHECK_EQUAL(traj.get_init_rotation(), init_rot);
```

**Pull request merged**